// Generated from echo_service.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.echo_service;

// Namespace: basic
// Basic echo operations

// Method: echo
// Echo back the input message
message echo_request {
  string message = 1;
}

message echo_response {
  string response = 1;
}

service echo_service {
  rpc echo(echo_request) returns (echo_response);
}

// Method: echo_with_delay
// Echo back the input message after a delay
message echo_with_delay_request {
  string message = 1;
  uint32 delay_ms = 2;
}

message echo_with_delay_response {
  string response = 1;
  uint32 actual_delay_ms = 2;
}

service echo_with_delay_service {
  rpc echo_with_delay(echo_with_delay_request) returns (echo_with_delay_response);
}

// Method: concat
// Concatenate two strings
message concat_request {
  string first = 1;
  string second = 2;
  string separator = 3;
}

message concat_response {
  string result = 1;
}

service concat_service {
  rpc concat(concat_request) returns (concat_response);
}


